go/libraries/doltcore/sqle/database.go

```diff
@@ -2213,14 +2315,26 @@ func (db Database) dropFragFromSchemasTable(ctx *sql.Context, fragType, name str
 		return missingErr
 	}
 
+	if resolve.UseSearchPath && schemaName == "" {
+		root, err := db.GetRoot(ctx)
+		if err != nil {
+			return err
+		}
+		schemaName, err = resolve.FirstExistingSchemaOnSearchPath(ctx, root)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
It's kinda gross IMO that I need to set `db.schemaName` to `dolt` above to get the `dolt.schemas` table, but then set it to the first existing schema here to get the view to drop. Option to suggestions